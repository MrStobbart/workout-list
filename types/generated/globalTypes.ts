/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

//==============================================================
// START Enums and Input Objects
//==============================================================

export enum WorkoutCategory_enum {
  c1 = "c1",
  c2 = "c2",
  c3 = "c3",
  c4 = "c4",
  c5 = "c5",
  c6 = "c6",
  c7 = "c7",
}

//==============================================================
// END Enums and Input Objects
//==============================================================
