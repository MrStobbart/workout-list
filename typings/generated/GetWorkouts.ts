/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: GetWorkouts
// ====================================================

export interface GetWorkouts_Workout {
  __typename: "Workout";
  id: Hasurauuid;
}

export interface GetWorkouts {
  /**
   * fetch data from the table: "Workout"
   */
  Workout: GetWorkouts_Workout[];
}
