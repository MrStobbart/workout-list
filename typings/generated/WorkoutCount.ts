/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: WorkoutCount
// ====================================================

export interface WorkoutCount_workout_aggregate_aggregate {
  __typename: "workout_aggregate_fields";
  count: number | null;
}

export interface WorkoutCount_workout_aggregate {
  __typename: "workout_aggregate";
  aggregate: WorkoutCount_workout_aggregate_aggregate | null;
}

export interface WorkoutCount {
  /**
   * fetch aggregated fields from the table: "workout"
   */
  workout_aggregate: WorkoutCount_workout_aggregate;
}
