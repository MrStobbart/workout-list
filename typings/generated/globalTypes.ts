/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

//==============================================================
// START Enums and Input Objects
//==============================================================

export enum workout_category_enum {
  c1 = "c1",
  c2 = "c2",
  c3 = "c3",
  c4 = "c4",
  c5 = "c5",
  c6 = "c6",
  c7 = "c7",
}

/**
 * input type for inserting data into table "workout"
 */
export interface workout_insert_input {
  category?: workout_category_enum | null;
  created_at?: Hasuratimestamptz | null;
  description?: string | null;
  id?: Hasurauuid | null;
  name?: string | null;
  start_date?: Hasuradate | null;
  updated_at?: Hasuratimestamptz | null;
}

//==============================================================
// END Enums and Input Objects
//==============================================================
