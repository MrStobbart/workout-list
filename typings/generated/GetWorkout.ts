/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { workout_category_enum } from "./globalTypes";

// ====================================================
// GraphQL query operation: GetWorkout
// ====================================================

export interface GetWorkout_workout {
  __typename: "workout";
  id: Hasurauuid;
  name: string;
  description: string;
  start_date: Hasuradate;
  category: workout_category_enum;
}

export interface GetWorkout {
  /**
   * fetch data from the table: "workout"
   */
  workout: GetWorkout_workout[];
}

export interface GetWorkoutVariables {
  workoutId: Hasurauuid;
}
