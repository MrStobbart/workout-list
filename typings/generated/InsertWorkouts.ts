/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { Workout_insert_input } from "./globalTypes";

// ====================================================
// GraphQL mutation operation: InsertWorkouts
// ====================================================

export interface InsertWorkouts_insert_Workout_returning {
  __typename: "Workout";
  id: Hasurauuid;
}

export interface InsertWorkouts_insert_Workout {
  __typename: "Workout_mutation_response";
  /**
   * data of the affected rows by the mutation
   */
  returning: InsertWorkouts_insert_Workout_returning[];
}

export interface InsertWorkouts {
  /**
   * insert data into the table: "Workout"
   */
  insert_Workout: InsertWorkouts_insert_Workout | null;
}

export interface InsertWorkoutsVariables {
  workouts: Workout_insert_input[];
}
