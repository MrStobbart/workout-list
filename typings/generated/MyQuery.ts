/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { WorkoutCategory_enum } from "./globalTypes";

// ====================================================
// GraphQL query operation: MyQuery
// ====================================================

export interface MyQuery_Workout {
  __typename: "Workout";
  description: string;
  startDate: Hasuradate;
  id: Hasurauuid;
  name: string;
  category: WorkoutCategory_enum;
}

export interface MyQuery {
  /**
   * fetch data from the table: "Workout"
   */
  Workout: MyQuery_Workout[];
}
