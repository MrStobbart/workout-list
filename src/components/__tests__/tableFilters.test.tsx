import React from "react";
import moment from "moment";
import renderer from "react-test-renderer";
import TableFilters, {
  monthIsNotWithinNextTwelveMonths,
} from "../tableFilters";
import { workout_category_enum } from "../../../typings/generated/globalTypes";

describe("monthIsNotWithinNextTwelveMonths", () => {
  it("return false for dates within the next twelve months", () => {
    expect(monthIsNotWithinNextTwelveMonths(moment().add(364, "days"))).toBe(
      false
    );
    expect(monthIsNotWithinNextTwelveMonths(moment().add(1, "days"))).toBe(
      false
    );
  });

  it("return true for dates in the past", () => {
    expect(
      monthIsNotWithinNextTwelveMonths(moment().subtract(1, "month"))
    ).toBe(true);
    expect(
      monthIsNotWithinNextTwelveMonths(moment().subtract(12, "years"))
    ).toBe(true);
  });

  it("return true for dates further than 1 year in the future ", () => {
    expect(monthIsNotWithinNextTwelveMonths(moment().add(13, "month"))).toBe(
      true
    );
    expect(monthIsNotWithinNextTwelveMonths(moment().add(2, "years"))).toBe(
      true
    );
  });
});

describe("tableFilter component", () => {
  it("renders correctly", () => {
    const setFilters = jest.fn();
    const component = renderer.create(
      <TableFilters
        filters={{
          page: 1,
          categories: [workout_category_enum.c1],
          month: moment(),
        }}
        setFilters={setFilters}
      />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
