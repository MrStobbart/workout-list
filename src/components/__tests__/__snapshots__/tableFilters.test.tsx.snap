// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tableFilter component renders correctly 1`] = `
<div
  className="ant-row"
  style={
    Object {
      "marginLeft": -8,
      "marginRight": -8,
    }
  }
>
  <div
    className="ant-col ant-col-8 gutter-row"
    style={
      Object {
        "paddingLeft": 8,
        "paddingRight": 8,
      }
    }
  >
    <div
      className="ant-picker"
      onMouseUp={[Function]}
      style={
        Object {
          "width": "135px",
        }
      }
    >
      <div
        className="ant-picker-input"
      >
        <input
          autoComplete="off"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          placeholder="Select month"
          readOnly={true}
          size={12}
          title=""
          value=""
        />
        <span
          className="ant-picker-suffix"
        >
          <span
            aria-label="calendar"
            className="anticon anticon-calendar"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="calendar"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
              />
            </svg>
          </span>
        </span>
      </div>
    </div>
  </div>
  <div
    className="ant-col ant-col-8 gutter-row"
    style={
      Object {
        "paddingLeft": 8,
        "paddingRight": 8,
      }
    }
  >
    <div
      className="ant-select ant-select-multiple ant-select-show-search"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      style={
        Object {
          "width": "350px",
        }
      }
    >
      <div
        className="ant-select-selector"
        onMouseDown={[Function]}
      >
        <span
          className="ant-select-selection-item"
        >
          <span
            className="ant-select-selection-item-content"
          >
            c1
          </span>
          <span
            aria-hidden={true}
            className="ant-select-selection-item-remove"
            onClick={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            <span
              aria-label="close"
              className="anticon anticon-close"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="close"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          className="ant-select-selection-search"
          style={
            Object {
              "width": 0,
            }
          }
        >
          <input
            aria-activedescendant="undefined_list_0"
            aria-autocomplete="list"
            aria-controls="undefined_list"
            aria-haspopup="listbox"
            aria-owns="undefined_list"
            autoComplete="off"
            className="ant-select-selection-search-input"
            onChange={[Function]}
            onCompositionEnd={[Function]}
            onCompositionStart={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onPaste={[Function]}
            readOnly={true}
            role="combobox"
            style={
              Object {
                "opacity": 0,
              }
            }
            unselectable="on"
            value=""
          />
          <span
            aria-hidden={true}
            className="ant-select-selection-search-mirror"
          >
            
            Â 
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
